src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxBase.TextBoxBaseUiaTextProviderTests.cs

```diff
@@ -0,0 +1,492 @@
+            UiaCore.SupportedTextSelection uiaTextRange = provider.SupportedTextSelection;
+            Assert.Equal(UiaCore.SupportedTextSelection.Single, uiaTextRange);
+        }
+
+        [WinFormsFact]
+        public void TextBoxBaseUiaTextProvider_GetCaretRange_IsNotNull()
+        {
+            using TextBoxBase textBoxBase = new SubTextBoxBase();
+            TextBoxBaseUiaTextProvider provider = new TextBoxBaseUiaTextProvider(textBoxBase);
+            UiaCore.ITextRangeProvider uiaTextRange = provider.GetCaretRange(out _);
+            Assert.NotNull(uiaTextRange);
+        }
+
+        [WinFormsFact]
+        public void TextBoxBaseUiaTextProvider_GetFirstVisibleLine_ReturnsCorrectValue()
+        {
+            using TextBoxBase textBoxBase = new SubTextBoxBase();
+            TextBoxBaseUiaTextProvider provider = new TextBoxBaseUiaTextProvider(textBoxBase);
+
+            int line = provider.FirstVisibleLine;
+            Assert.Equal(0, line);
+
+            textBoxBase.Multiline = true;
+
+            textBoxBase.Size = new Size(30, 100);
+            line = provider.FirstVisibleLine;
+            Assert.Equal(0, line);
+        }
+
+        [WinFormsFact]
+        public void TextBoxBaseUiaTextProvider_GetLineCount_ReturnsCorrectValue()
+        {
+            using TextBoxBase textBoxBase = new SubTextBoxBase();
+            TextBoxBaseUiaTextProvider provider = new TextBoxBaseUiaTextProvider(textBoxBase);
+            Assert.Equal(1, provider.LinesCount);
+
+            textBoxBase.Multiline = true;
+            textBoxBase.Size = new Size(30, 50);
+            Assert.Equal(1, provider.LinesCount);
+
+            textBoxBase.Text += "1\r\n";
+            Assert.Equal(2, provider.LinesCount);
+
+            textBoxBase.Text += "2\r\n";
+            Assert.Equal(3, provider.LinesCount);
+
+            textBoxBase.Text += "3\r\n";
+            Assert.Equal(4, provider.LinesCount);
+
+            textBoxBase.Text += "4\r\n";
+            Assert.Equal(5, provider.LinesCount);
+        }
+
+        public static IEnumerable<object[]> TextBoxBase_GetLineFromCharIndex_TestData()
+        {
+            yield return new object[] { new SubTextBoxBase { Size = new Size(50, 20), Multiline = false }, 0, 0 };
+            yield return new object[] { new SubTextBoxBase { Size = new Size(50, 20), Multiline = false }, 50, 0 };
+            yield return new object[] { new SubTextBoxBase { Size = new Size(100, 50), Multiline = true }, 50, 3 };
+            yield return new object[] { new SubTextBoxBase { Size = new Size(50, 50), Multiline = true }, 50, 8 };
+        }
+
+        [WinFormsTheory]
+        [MemberData(nameof(TextBoxBase_GetLineFromCharIndex_TestData))]
+        public void TextBoxBaseUiaTextProvider_GetLineFromCharIndex_ReturnsCorrectValue(TextBoxBase textBoxBase, int charIndex, int expectedLine)
+        {
+            textBoxBase.Text = "Some test text for testing GetLineFromCharIndex method";
```

Comment:
Don't pass controls as theory data where possible
```suggestion
        public static IEnumerable<object[]> TextBoxBase_GetLineFromCharIndex_TestData()
        {
            yield return new object[] { new Size(50, 20), false, 0, 0 };
            // ...
        }

        [WinFormsTheory]
        [MemberData(nameof(TextBoxBase_GetLineFromCharIndex_TestData))]
        public void TextBoxBaseUiaTextProvider_GetLineFromCharIndex_ReturnsCorrectValue(Size size, bool multiline, int charIndex, int expectedLine)
        {
            using var textBoxBase = new SubTextBoxBase { Size = size, Multiline = multiline };
            textBoxBase.Text = "Some test text for testing GetLineFromCharIndex method";
```

Applies universally